### PR TITLE
planner: Adjust the log level and returned value when `cacheableChecker` check `*ast.TableName` nodes 

### LINKED ISSUE #46760: Adjust the log level and returned value when `cacheableChecker` check `*ast.TableName` nodes
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. If calling `TableByName` encounters an error, we need to adjust the log level and supplement the error log. Because this error does not affect execution results.

https://github.com/pingcap/tidb/blob/41d1ec0267efae5c3263e3f611025c82c14cd177/planner/core/plan_cacheable_checker.go#L557-L609

2. Besides, if we encounter this error when calling `TableByName`, I think it makes sense to return true

### 2. What did you expect to see? (Required)
Change this log to a warning and add some info to check the cause of this error later.

### 3. What did you see instead (Required)
`logutil.BgLogger().Error("Error occur in checking cacheable", zap.Error(err))`

### 4. What is your TiDB version? (Required)

Release Version: v7.4.0-alpha-292-g5cbcc86e50
Edition: Community
Git Commit Hash: 5cbcc86e508da1af91112ad42a07b9863a61185c

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #46760

Problem Summary: planner: Adjust the log level and returned value when `cacheableChecker` check `*ast.TableName` nodes 

### What is changed and how it works?

planner: Adjust the log level and returned value when `cacheableChecker` check `*ast.TableName` nodes 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: planner/core/plan_cacheable_checker.go [hunk 1]
```diff
@@ -185,7 +197,14 @@ func (checker *cacheableChecker) Enter(in ast.Node) (out ast.Node, skipChildren
 		}
 	case *ast.TableName:
 		if checker.schema != nil {
-			if isPartitionTable(checker.schema, node) {
+			isPartitioned, err := isPartitionTable(checker.schema, node)
+			if err != nil {
+				checker.cacheable = false
+				checker.reason = fmt.Errorf("check partition table failed: %w", err).Error()
+				logutil.BgLogger().Warn("check partition table failed", zap.Error(err))
```
Comment (zimulala) [inline planner/core/plan_cacheable_checker.go:204]: Do we need to add SQL info to this log?

Comment (qw4990) [reply planner/core/plan_cacheable_checker.go]: Updated, PTAL

File: planner/core/plan_cacheable_checker.go [hunk 2]
```diff
@@ -185,7 +197,15 @@ func (checker *cacheableChecker) Enter(in ast.Node) (out ast.Node, skipChildren
 		}
 	case *ast.TableName:
 		if checker.schema != nil {
-			if isPartitionTable(checker.schema, node) {
+			isPartitioned, err := isPartitionTable(checker.schema, node)
+			if err != nil {
+				checker.cacheable = false
+				checker.reason = fmt.Errorf("check partition table failed: %w", err).Error()
+				logutil.BgLogger().Warn("check partition table failed", zap.Error(err),
+					zap.String("sql", trimIfTooLong(checker.sctx.GetSessionVars().StmtCtx.OriginalSQL)))
```
Comment (zimulala) [inline planner/core/plan_cacheable_checker.go:205]: Do we also need to add these updates to `nonPreparedPlanCacheableChecker`?

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**